api/routes.go

```diff
@@ -0,0 +1,194 @@
+package api
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/julienschmidt/httprouter"
+)
+
+// buildHttpRoutes sets up and returns an * httprouter.Router.
+// it connected the Router to the given api using the required
+// parameters: requiredUserAgent and requiredPassword
+func buildHttpRoutes(api *API, requiredUserAgent string, requiredPassword string) *httprouter.Router {
```

Comment:
make this a function of the API instead of passing in the api as an argument, otherwise it's not clear that the state of the api is being modified

Reply:
Alternatively, we could change this func to be stateless and return the router, and assign the API's `router` to the value returned by the func where the API is initialized.

Reply:
all good points and either one is a step up in terms of clarity but i would not know which is better.

Reply:
Needs to be a method of the api I believe because when you build the routes you need to understand which modules have been initialized.